Assets/Scripts/WeaponSystem/HandCannon.cs

```diff
@@ -74,7 +74,18 @@ public void Shoot()
 						Destroy(impact, impact.TryGetComponent(out ParticleSystem particles) ? particles.main.duration : 2f);
 					}
 
-					//TODO: Deal damage
+					var enemy = hit.collider.GetComponentInParent<EnemyData>();
+					if (enemy != null) // We hit an ennemy!
+					{
+						var finalDamage = enemy.TakeDamage(hit.collider, 10);
+						if (_damageDisplay != null)
+						{
+							var go = Instantiate(_damageDisplay, hit.point, Quaternion.identity);
+							go.GetComponent<TMP_Text>().text = finalDamage.ToString();
+							go.transform.LookAt(player.transform.position, Vector3.up);
+							go.transform.rotation = Quaternion.Euler(0f, go.transform.rotation.eulerAngles.y + 180, 0f);
+						}
```

Comment:
I suggest having the damage display on its own system. The font will have to be consistent, font size, way it behaves, etc. It could be pretty simple with such API:
```
static event Action<float, Vector3> OnDamageDisplayRequested;
DamageDisplay.NotifyDamage(float damage, Vector3 location);
```
and an **optional** display component in a UI scene:
```
DamageDisplay.OnDamageDisplayRequested += HandleDamageDisplay;
```

Reply:
Damage display was mostly for debug, don't think we will let it in the final game